cmd/tailscale/cli/serve_dev_test.go

```diff
@@ -838,11 +837,99 @@ func TestSrcTypeFromFlags(t *testing.T) {
 				t.Errorf("Expected error: %v, got: %v", tt.expectedErr, err)
 			}
 			if srcType != tt.expectedType {
-				t.Errorf("Expected srcType: %s, got: %s", tt.expectedType, srcType)
+				t.Errorf("Expected srcType: %s, got: %s", tt.expectedType.String(), srcType)
 			}
 			if srcPort != tt.expectedPort {
 				t.Errorf("Expected srcPort: %d, got: %d", tt.expectedPort, srcPort)
 			}
 		})
 	}
 }
+
+func TestExpandProxyTargetDev(t *testing.T) {
+	tests := []struct {
+		input         string
+		expected      string
+		expectedError string
```

Comment:
let's not test on error strings as that will be too fragile, a simple re-wording of an error message will cause failures. Instead, let's just test make `wantErr bool` which checks if a certain `input` should fail